File: src/problem025/solution025.py
def fibs():
    n, m = 1, 1
    while True:
        yield n
        n, m = m, n + m


def main():
    for n, fib in enumerate(fibs(), 1):
        if fib >= 10 ** 999:
            print(n)
            break

if __name__ == '__main__':
    main()

File: src/problem025/solution025.hs
import Data.List (zipWith)
main = print $ n
     where (n, fib) = head . filter (\(n, fib) -> fib >= 10 ^ 999) . zip [1..] $ fibs
fibs = 1 : 1 : zipWith (+) fibs (tail fibs)

